### ISSUE TITLE
Build failing while following youtube example : youtube8m:extract_yt8m_features Failed

### ISSUE THREAD

**Issue by Tortoise17** [google/mediapipe#3642]
Dear colleagues, since, I am struggling to make the user example run, 
I have tried several remedies available on git and stackoverflow. 

Still I am facing this issue below 
command used 
`
 bazel build -c opt --linkopt=-s  --define MEDIAPIPE_DISABLE_GPU=1 --define no_aws_support=true  mediapipe/examples/desktop/youtube8m:extract_yt8m_features`

```
ERROR: /home/user/.cache/bazel/_bazel_user/7750f0c8784028cd1fdf8f62c71a9a8a/external/com_google_protobuf/BUILD:515:10: Compiling src/google/protobuf/compiler/main.cc failed: undeclared inclusion(s) in rule '@com_google_protobuf//:protoc':
this rule is missing dependency declarations for the following files included by 'src/google/protobuf/compiler/main.cc':
  '/usr/local/lib/gcc/x86_64-pc-linux-gnu/10.4.0/include/stdarg.h'
  '/usr/local/lib/gcc/x86_64-pc-linux-gnu/10.4.0/include/stddef.h'
  '/usr/local/lib/gcc/x86_64-pc-linux-gnu/10.4.0/include/stdint.h'
Target //mediapipe/examples/desktop/youtube8m:extract_yt8m_features failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2.317s, Critical Path: 2.07s
INFO: 129 processes: 129 internal.
FAILED: Build did NOT complete successfully
```

I am using 
CentOS 7
gcc 10.4.0
cuda 11.2
tensorflow 2.9.1 (tried with and without gpu) same error
OpenCV 4.6.0-dev
bazel 5.2.0

If there is any supporting help which can assist me identifying what mistake I am doing in reaching this target run. 
Your kind help and advise will be highly appreciated

**Comment by kuaashish** [comment]
Hi @Tortoise17, Please refer to this closed [issue](https://github.com/google/j2cl/issues/21) and [comment](https://github.com/google/j2cl/issues/21#issuecomment-525527002) for your query. Thank you!

**Comment by Tortoise17** [comment]
@sureshdagooglecom @kuaashish @cdibona @camillol  dear friends. Can you guess now what error could be and reason for this failure? 
```

ERROR: /run/media/user/soccer/work_/mediapipe/mediapipe/examples/desktop/youtube8m/BUILD:17:10: Linking mediapipe/examples/desktop/youtube8m/extract_yt8m_features failed: (Exit 1): gcc failed: error executing command /usr/local/bin/gcc @bazel-out/k8-opt/bin/mediapipe/examples/desktop/youtube8m/extract_yt8m_features-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
bazel-out/k8-opt/bin/mediapipe/calculators/tensorflow/_objs/pack_media_sequence_calculator/pack_media_sequence_calculator.o:pack_media_sequence_calculator.cc:function mediapipe::PackMediaSequenceCalculator::Process(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::allocate(unsigned long)'
bazel-out/k8-opt/bin/mediapipe/calculators/tensorflow/_objs/pack_media_sequence_calculator/pack_media_sequence_calculator.o:pack_media_sequence_calculator.cc:function mediapipe::PackMediaSequenceCalculator::Process(mediapipe::CalculatorContext*): error: undefined reference to 'cv::imencode(cv::String const&, cv::_InputArray const&, std::vector<unsigned char, std::allocator<unsigned char> >&, std::vector<int, std::allocator<int> > const&)'
bazel-out/k8-opt/bin/mediapipe/calculators/tensorflow/_objs/pack_media_sequence_calculator/pack_media_sequence_calculator.o:pack_media_sequence_calculator.cc:function mediapipe::PackMediaSequenceCalculator::Process(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::deallocate()'
bazel-out/k8-opt/bin/mediapipe/calculators/tensorflow/_objs/pack_media_sequence_calculator/pack_media_sequence_calculator.o:pack_media_sequence_calculator.cc:function mediapipe::PackMediaSequenceCalculator::Process(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::deallocate()'
bazel-out/k8-opt/bin/mediapipe/calculators/video/_objs/opencv_video_decoder_calculator/opencv_video_decoder_calculator.o:opencv_video_decoder_calculator.cc:function mediapipe::OpenCvVideoDecoderCalculator::Open(mediapipe::CalculatorContext*): error: undefined reference to 'cv::VideoCapture::VideoCapture(cv::String const&)'
bazel-out/k8-opt/bin/mediapipe/calculators/video/_objs/opencv_video_decoder_calculator/opencv_video_decoder_calculator.o:opencv_video_decoder_calculator.cc:function mediapipe::OpenCvVideoDecoderCalculator::Open(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::deallocate()'
bazel-out/k8-opt/bin/mediapipe/calculators/video/_objs/opencv_video_decoder_calculator/opencv_video_decoder_calculator.o:opencv_video_decoder_calculator.cc:function mediapipe::OpenCvVideoDecoderCalculator::Open(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::allocate(unsigned long)'
bazel-out/k8-opt/bin/mediapipe/calculators/video/_objs/opencv_video_decoder_calculator/opencv_video_decoder_calculator.o:opencv_video_decoder_calculator.cc:function mediapipe::OpenCvVideoDecoderCalculator::Open(mediapipe::CalculatorContext*): warning: the use of `tmpnam' is dangerous, better use `mkstemp'
bazel-out/k8-opt/bin/mediapipe/calculators/video/_objs/opencv_video_decoder_calculator/opencv_video_decoder_calculator.o:opencv_video_decoder_calculator.cc:function mediapipe::OpenCvVideoDecoderCalculator::Open(mediapipe::CalculatorContext*): error: undefined reference to 'cv::String::deallocate()'
collect2: error: ld returned 1 exit status
Target //mediapipe/examples/desktop/youtube8m:extract_yt8m_features failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 972.060s, Critical Path: 360.72s
INFO: 4208 processes: 114 internal, 4094 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by kuaashish** [comment]
Hi @Tortoise17, Please go through this closed issue #666 and could you try the steps mentioned in the [#666(comment)](https://github.com/google/mediapipe/issues/666#issuecomment-620884783) for your latest query. Thank you!
